docs/backend/backend-overview.md

```diff
@@ -0,0 +1,210 @@
+  correctly on the cluster is the concern of the Backend Service Owner.
+
+If we map this back to the InstructLab project itself, you can imagine that the
+Backend Maintainers team serves as Backend Service Owners. There may be another
+team, the taxonomy maintainers, that represent InstructLab Model Developers,
+focused on evaluating taxonomy contributions and running batches of them through
+the backend pipeline.
+
+![High-Level Service Diagram](./images/backend-service-high-level.png)
+
+### Observable
+
+The Backend Service must be observable, both by the Backend Service Owners and
+by InstructLab Model Developers.
+
+### Reusable
+
+The Backend Service was envisioned from the beginning as a pipeline of distinct
+steps (data generation, training, evaluation, and publishing). This separation
+of concerns should be maintained in a future service-oriented architecture.
+The diagram under [Service-Oriented](#service-oriented) shows a high-level diagram
+with a single "InstructLab Service API." Implementation of each sub-section of
+the functionality should be done in such a way that the core of it is reusable.
+In some cases that may be via libraries. In others, it may make sense to build a
+microservice that can be used in other contexts.
+
+Within InstructLab, we could have a top-level application that implements the
+entire backend pipeline workflow and does so by making use of each of these
+libraries and/or services.
+
+![Service Breakdown](./images/backend-service-breakdown.png)
+
+### Small Iterations
+
+While we may cast a vision for the future, we desire to get there via a series
+of small iterations. We have a working backend pipeline already and will
+continue to run it every week. In parallel, we can work to evolve it through a
+series of small iterations that evolve our processes over time to use new
+service-based interfaces, gradually evolving the execution process that occurs
+each week.
+
+We do not want a new effort happening in parallel to what is already working
+today.
+
+### Minimize Mandatory External Dependencies
+
+Major external dependencies should be made optional where feasible to do so.
+The more of these we require, the more complicated it will be to reuse this
+system in other environments.
+
+In other cases, it may be appropriate to provide an abstraction for the
+dependency so that alternatives can be put in place. Alternate Object Storage
+systems are a good example.
+
+### Make Development Accessible
+
+Most people will not have direct access to a large cluster to test changes
+to this system. We should create development environments and minimal
+configurations as much as possible to lower the barrier to doing useful
+development and testing with these components.
+
+### Model Independence
+
+We desire to keep the backend flexible such that serving and training alternate
+models in the future is not more work than absolutely necessary.
+
+## Key Technology Decisions
```

Comment:
Kubeflow is a strategic priority for OpenShift AI. Do you think it would be among one of the acceptable dependencies? 

For larger scale fine tuning and orchestrating more intensive pipelines, the [Training Operator](https://www.kubeflow.org/docs/components/training/overview/) and [KFP](https://www.kubeflow.org/docs/components/pipelines/overview/) feel like they could be key technology decisions. 

Any thoughts here @gabe-l-hart?

Reply:
This is definitely interesting. I believe at this point for kubernetes-based backend deployments (which notably does not include RHEL AI), the Training Operator will almost certainly be a key piece of technology. KFP may also play a role in data prep if/when the SDG workflows expand to include code-based customization versus just config-based (the current implementation doesn't allow this, but it may happen in the future).

I haven't taken a pass over this doc in a while and I know it was originally written before the decision was final to not include a lightweight kubernetes distro in RHEL AI (see discussion below). @russellb do you plan to update the doc based on the state of kube as a prereq now that we're further along in the RHEL AI process?

Reply:
I don't plan to update this right now as I have some other priorities.